cmd/pebble/badger.go

```diff
@@ -0,0 +1,216 @@
+// Copyright 2018 The LevelDB-Go and Pebble Authors. All rights reserved. Use
+// of this source code is governed by a BSD-style license that can be found in
+// the LICENSE file.
+
+//go:build badger
+// +build badger
+
+package main
+
+import (
+	"bytes"
+	"expvar"
+	"log"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/cockroachdb/pebble/internal/cache"
+	"github.com/cockroachdb/pebble/sstable"
+	"github.com/outcaste-io/badger/v3"
+)
+
+// Adapters for Badger.
+type badgerDB struct {
+	db *badger.DB
+}
+
+func newBadgerDB(dir string) DB {
+	db, err := badger.Open(badger.DefaultOptions(dir))
```

Comment:
TODO

+	// Badger iterator has a prefix setting for opts. That would selectively choose only the tables
+	// which are relevant before doing a seek. So, seek would be faster.